### PR TITLE
Add Device Trust Web proto definitions

### PR DESCRIPTION
**Pull request by codingllama**
Add/modify protos for the Device Trust Web implementation.

The RFD is still in review, but I'm getting a jump start in coding. As usual, I'll make further PRs if any of this changes during review.

See https://github.com/gravitational/teleport.e/blob/rfd/0009e-device-trust-web-support/rfd/0009e-device-trust-web-support.md#proto-specification.

https://github.com/gravitational/teleport.e/issues/3236

### CODE REVIEW

File: proto/teleport/lib/teleterm/v1/service.proto [hunk 1]
```diff
@@ -163,6 +164,13 @@ service TerminalService {
   // UpdateUserPreferences updates the preferences for a given user in appropriate root and leaf clusters.
   // Only the properties that are set (cluster_preferences, unified_resource_preferences) will be updated.
   rpc UpdateUserPreferences(UpdateUserPreferencesRequest) returns (UpdateUserPreferencesResponse);
+
+  // AuthenticateWebDevice blesses a web session with device trust by performing
+  // the on-behalf-of device authentication ceremony.
+  //
+  // See
+  // https://github.com/gravitational/teleport.e/blob/master/rfd/0009e-device-trust-web-support.md#device-web-authentication.
+  rpc AuthenticateWebDevice(AuthenticateWebDeviceRequest) returns (AuthenticateWebDeviceResponse);
```
Comment (codingllama) [inline proto/teleport/lib/teleterm/v1/service.proto:173]: FYI @ravicious, I'm getting a jumpstart in the proto definitions here.

**Comment by codingllama** [comment]
Line count is wild, but it's because of generated protos. The actual change is just a handful of new messages.

**Comment by codingllama** [comment]
Friendly ping reviewers? @gabrielcorado @ryanclark ?

**Review by ryanclark — APPROVED** [review]

**Review by avatus — APPROVED** [review]

**Comment by codingllama** [comment]
Rebased onto master, protogen conflicts solved, squashed all protogen commits together. Thanks for the reviews.

**Pull request merged**